### ISSUE TITLE
Elements inside shadow dom are hidden by its host (causing them to not be interactable)

### ISSUE THREAD

**Issue by TomCaserta** [cypress-io/cypress#18008]
### Current behavior

Currently when trying to click an element inside slotted shadow DOM Cypress incorrectly believes the element is covered by its host element.

```
Timed out retrying after 4050ms: cy.click() failed because this element:

<button data-test-id="my-button">...</button>

is being covered by another element:

<my-custom-element id="element">Click me!</my-custom-element>

Fix this problem, or use {force: true} to disable error checking.Learn more
```

### Desired behavior

Cypress should correctly be able to figure out if the shadow dom element is a descendent of the element that is covering it.

### Test code to reproduce

https://github.com/TomCaserta/cypress-shadow-reproduction 

You can find all the test case code in the `index.html` file (which can be run via `npm run start`). The failing tests can be found in the `error.spec.js` file.

### Cypress Version

8.3.1

### Other

All cypress versions prior to 8.3.1 (and after Shadow DOM support was added) have been problematic.

**Comment by TomCaserta** [comment]
Note: I encountered this issue when migrating a large web component codebase from my fork of Cypress to the latest version of Cypress. I had already added Shadow DOM support there and as such have a large amount of tests I can try any solution out against. 

I will try in the coming week to tackle this myself and contribute it back if I'm able to.

**Comment by sainthkh** [comment]
The "covering" problem will be fixed by #18908. But there are still some behavior differences because events are not sent to elements inside shadow roots. 

It'll be fixed with #17531.
